public/friendlycode/js/fc/ui/bramble-proxy.js

```diff
@@ -0,0 +1,95 @@
+// This file is the friendlycode half of a proxy layer between Thimble's live preview
+// window and bramble's codemirror instance. It was designed to map to existing
+// friendlycode use of codemirror, with the exception of duplicate functionality
+// between bramble and friendlycode that bramble already provides (or will)
+define(["backbone-events"], function(BackboneEvents) {
+  "use strict";
+
+  var eventCBs = {
+    "change": [],
+    "reparse": []
+  };
+
+  function BrambleProxy(place, givenOptions) {
+    var iframe = document.createElement("iframe");
+    var latestSource = "(none)";
+    var telegraph;
+
+    // Event listening for proxied event messages from our editor iframe.
+    window.addEventListener("message", function(evt) {
+      // Set the communication channel to our iframe
+      // now that it's signaled that it has content
+      // by sending a postMessage
+      telegraph = iframe.contentWindow;
+      var message = JSON.parse(evt.data);
+      if (typeof message.type !== "string" || message.type.indexOf("bramble") === -1) {
+        return;
+      }
+      if (message.type == "bramble:change") {
+        latestSource = message.sourceCode;
+        eventCBs["change"].forEach(function(cb){
+          cb();
+        });
+        return;
+      }
+      if (message.type == "bramble:init") {
+        telegraph.postMessage(JSON.stringify({
+          type: "bramble:init",
+          source: latestSource
+        }), "*");
+      }
+    });
+
+    // Create CodeMirror-like interface for friendlycode to use
+    this.getValue = function() {
+      return latestSource;
```

Comment:
Do we actually still need to do all this still?  The only reason for this was so that the Thimble preview could access the source, but we don't do that now.  Why are we still exposing/consuming CodeMirror type stuff in Thimble?